### PR TITLE
fix: event saving without dance-style and required validations added

### PR DESCRIPTION
**Pull request by amanuela97**
## Changes Made
- Can save an event without dance style
- A message is shown when the required fields name and startDate are not filled

## How to test
- Open https://wedance-amanuela97.netlify.app/events/_id/edit
- Save a new event without filling in name or start date and it should show an error message
- Try saving a new event without a dance style and it should work

Demo Link: https://wedance-amanuela97.netlify.app/

### CODE REVIEW

File: pages/events/_id/edit.vue [hunk 1]
```diff
@@ -17,6 +17,7 @@
         vertical
         :show-remove="!!item.id"
         :show-copy="!!item.id"
+        :event-error="eventError"
```
Comment (razbakov) [inline pages/events/_id/edit.vue:20]: Form component shouldn't know about event

**Comment by razbakov** [comment]
If I scroll bottom and click save, I don't see error message. Try to use $toaster instead.

**Review by razbakov — CHANGES_REQUESTED** [review]

File: pages/events/_id/edit.vue (hunk 1 above)
Comment (best-lucky1030) [reply pages/events/_id/edit.vue:20]: @razbakov 
What does it mean?
`form component shouldn't know about event`

**Review by best-lucky1030 — COMMENTED** [review]
put my request.

File: pages/events/_id/edit.vue (hunk 1 above)
Comment (razbakov) [reply pages/events/_id/edit.vue:20]: `TForm` is a component that renders forms. 
Event is a data structure.

Abstract component that renders forms should operate with abstract concepts like fields, errors, etc.

TForm component is used in Profile Editor, Event Editor, Post Editor. Following naming logic with `event-error` we would need to add also `profile-error`, `post-error`. That solution is not scalable.

File: pages/events/_id/edit.vue [hunk 2]
```diff
@@ -104,20 +104,32 @@ export default {
...
+          error = {
+            message: 'name is a required field!',
+            field: 'name',
+          }
+          this.errors.push(error)
+          this.$toast.error(error.message)
         }
-        return
+
+        if (!data.startDate) {
```
Comment (best-lucky1030) [inline pages/events/_id/edit.vue:123]: i think it's better to use `if` than `else if`
how about you?

**Review by best-lucky1030 — COMMENTED** [review]
suggestion.

**Pull request closed without merging**